internal/service/vpc/nat_gateway.go

```diff
@@ -138,83 +139,79 @@ func (n *natGatewayResource) Configure(_ context.Context, req resource.Configure
 
 func (n *natGatewayResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
 	var plan natGatewayResourceModel
-	diags := req.Plan.Get(ctx, &plan)
-	resp.Diagnostics.Append(diags...)
-	if resp.Diagnostics.HasError() {
-		return
-	}
 
 	if !n.config.SupportVPC {
 		resp.Diagnostics.AddError(
-			"Not support classic",
-			fmt.Sprintf("resource %s does not support classic", req.Config.Schema.Type().String()),
+			"NOT SUPPORT CLASSIC",
+			"resource does not support CLASSIC. only VPC.",
 		)
 		return
 	}
 
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
 	reqParams := &vpc.CreateNatGatewayInstanceRequest{
 		RegionCode: &n.config.RegionCode,
 		VpcNo:      plan.VpcNo.ValueStringPointer(),
 		ZoneCode:   plan.Zone.ValueStringPointer(),
-		SubnetNo:   plan.SubnetNo.ValueStringPointer(),
 	}
 
-	if !plan.Name.IsNull() {
+	if !plan.SubnetNo.IsNull() && !plan.SubnetNo.IsUnknown() {
+		reqParams.SubnetNo = plan.SubnetNo.ValueStringPointer()
+	} else {
+		resp.Diagnostics.AddError("CREATING ERROR", "subnet_no is required when creating a new NATGW")
+		return
+	}
```

Comment:
This is a trivial matter, but using early return would make it a little cleaner than if/else.
```go
if plan.SubnetNo.IsNull() || plan.SubnetNo.IsUnknown() {
	resp.Diagnostics.AddError("CREATING ERROR", "subnet_no is required when creating a new NATGW")
	return
}
reqParams.SubnetNo = plan.SubnetNo.ValueStringPointer()
```

Reply:
OK, changed.